Extensions/auto_tagger.js

```diff
@@ -289,6 +301,18 @@ XKit.extensions.auto_tagger = new Object({
 
 		}
 
+
+		if (XKit.extensions.auto_tagger.preferences.tag_source.value && obj.source_owner !== "") {
+			if (XKit.extensions.auto_tagger.preferences.tag_person_replace_hyphens.value) {
+				if (obj.source_owner !== undefined) {
```

Comment:
Right now, the behavior is very confusing: if you have hyphen replacement selected, you'll never see `source: undefined`. But if you have it disabled, you will see `source: undefined`. This is a sharp corner edge-case wise. Can you refactor this code so that we're only checking for the presence of source_owner once, and we only have one call to `mreturn_add`? Thanks!